### PR TITLE
Refine UG with more explanations and examples

### LINKED ISSUE #233: Missing exit action on search mode in command summary table

### PR DESCRIPTION
**Pull request by cshao02**
Fixes #233, #247, #252, #306

The UG now contains more explanations for the commands so that users will have less ambuiguity regarding the behaviour of our app.

Additionally, the message usage for event-add command is corrected where before it says that it only accepts one contact index which is not correct (a bug), now it say that it can accept one or more contact indices.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -117,43 +117,22 @@ Adds a contact to the address book.
...
 
+* Telegram usernames must start with an alphabet, contain only alphanumeric characters or underscores and be between 5 and 32 characters long.
+* The roles must be one of the following: attendee, vendor, volunteer and sponsor.
+
 <div markdown="span" class="alert alert-primary">:memo: **Note:**
 A contact can have any number of roles (including 0), and the Telegram username is optional. 
-The roles must be one of the following: attendee, vendor, volunteer and sponsor.
-Telegram usernames must start with an alphabet.
+
+:memo: **Note:** You will be unable to add a contact to PlanPal if there already exists a contact with the same phone number, email or telegram username.
```
Comment (brianchoon) [inline docs/UserGuide.md:126]: This is useful and addresses the ambiguity surrounding `Person` equality in our app!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -117,43 +117,22 @@ Adds a contact to the address book.
...
 </div>
 
 Examples:
 * `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
-* `add n/Betsy Crowe e/bc@gmail.com a/New Street p/1234567 t/betsyyy r/sponsor`
+* `add n/Betsy Crowe e/bc@gmail.com a/New Street p/1234567 t/betsyyy r/sponsor r/vendor`
```
Comment (brianchoon) [inline docs/UserGuide.md:131]: I like this detail, to illustrate that there can be more than one role!

**Review by brianchoon — APPROVED** [review]
LGTM!

**Review by chuajunyu — APPROVED** [review]
LGTM!

**Pull request merged**